components/wmtk_components/embedding/internal/Embedding.cpp

```diff
@@ -0,0 +1,152 @@
+#include "Embedding.hpp"
+#include "igl/writeOBJ.h"
+
+namespace wmtk::components::internal {
+Embedding::Embedding(
+    Eigen::Matrix<long, -1, -1>& m_edges_,
+    Eigen::MatrixXd& m_vertices_,
+    EmbeddingOptions& options_,
+    double m_blank_rate_
+    // double m_resolute_area_)
+    )
+    : m_edges(m_edges_)
+    , m_vertices(m_vertices_)
+    , options(options_)
+    , m_blank_rate(m_blank_rate_)
+//, m_resolute_area(m_resolute_area_)
+{
+    m_marked_vertices.reserve(m_vertices.rows());
+    m_marked_edges.reserve(m_edges.rows());
+
+    for (size_t i = 0; i < m_vertices.rows(); ++i) {
+        m_marked_vertices.emplace_back(i);
+    }
+    for (size_t i = 0; i < m_edges.rows(); ++i) {
+        m_marked_edges.emplace_back(std::pair<long, long>(m_edges(i, 0), m_edges(i, 1)));
+    }
+}
+void Embedding::compute_bounding_value(double& max_x, double& max_y, double& min_x, double& min_y)
+{
+    max_x = max_y = std::numeric_limits<double>::lowest();
+    min_x = min_y = std::numeric_limits<double>::max();
+    for (size_t i = 0; i < m_vertices.rows(); ++i) {
+        max_x = std::max(max_x, m_vertices(i, 0));
+        max_y = std::max(max_y, m_vertices(i, 1));
+        min_x = std::min(min_x, m_vertices(i, 0));
+        min_y = std::min(min_y, m_vertices(i, 1));
+    }
+    const double blank_region_length_x = (max_x - min_x) * m_blank_rate;
+    const double blank_region_length_y = (max_y - min_y) * m_blank_rate;
+    max_x += blank_region_length_x;
+    min_x -= blank_region_length_x;
+    max_y += blank_region_length_y;
+    min_y -= blank_region_length_y;
+}
+void Embedding::process()
+{
+    // find boundingbox
+    double max_x, min_x, max_y, min_y;
+    compute_bounding_value(max_x, max_y, min_x, min_y);
+
+    Eigen::MatrixXd bounding_box_vertices(4, 2);

```

Comment:
This has static size, so you can use `Eigen::Matrix<double, 4, 2>`.
I hope I got the syntax right but my main point is that you should avoid using `Eigen::MatrixXd` whenever possible.